### PR TITLE
[DEVELOPER-1795] Search Page

### PR DESCRIPTION
**Pull request by wesbos**

### CODE REVIEW

**Comment by RubyGemTSL** [comment]
@wesbos @paulrobinson - seems there is still an issue with the DCP as not getting any search results.

**Comment by paulrobinson** [comment]
This issue is that the DCP is returning zero results: http://10.19.112.67:57936/v2/rest/search?agg=per_project_counts&agg=tag_cloud&agg=top_contributors&agg=activity_dates_histogram&agg=per_sys_type_counts&field=sys_url_view&field=sys_title&field=sys_last_activity_date&field=sys_description&field=sys_tags&from=0&query=cdi&size=5&sortBy=new&type=rht_website

The equivalent on dcp2.jboss.org does return results: http://dcp2.jboss.org/v2/rest/search?agg=per_project_counts&agg=tag_cloud&agg=top_contributors&agg=activity_dates_histogram&agg=per_sys_type_counts&field=sys_url_view&field=sys_title&field=sys_last_activity_date&field=sys_description&field=sys_tags&from=0&query=cdi&size=5&sortBy=new&type=rht_website

I'll investigate why the PR staging instance of the DCP is missing this data.

**Comment by wesbos** [comment]
If it gives zero results, it should still say that.  I just see a broken URL on the IP address one

**Comment by paulrobinson** [comment]
http://10.19.112.67:57936 was accessible when I tried, it's not now. @RubyGemTSL is the cleanup of DCP servers still happening?

**Comment by RubyGemTSL** [comment]
its just a continuous loading spinner, is that what you see? 

@paulrobinson if the cleanup of DCP servers is being cleaned it's not from the acceptance tests as @robpblake commented this line out of last night.

**Comment by paulrobinson** [comment]
@RubyGemTSL are sure the updated code is being executed here? We've seen issues in the past where old code is being ran.

**Comment by RubyGemTSL** [comment]
@paulrobinson the site build looks like it has wes's latest changes as the search page URL is updated:
http://developers-pr.stage.redhat.com/pr/1086/build/1125/search/?q=containers

As for the tests failing ignore them, they would fail as they have been rewritten. 

The DCP being cleaned was a step in the jenkins job, it did not require any changes to code except in the jenkins job. That step is being ignored now so it should have nothing to do with that.

**Comment by RubyGemTSL** [comment]
http://10.19.112.67:57936/ is also not accessible for me

**Comment by robpblake** [comment]
@RubyGemTSL @paulrobinson It looks like the test job is called docker-compose stop and docker-compose down which will be deleting all of the containers started by the original pull request.

I'll try and work out where that is happening.

**Comment by robpblake** [comment]
@paulrobinson @RubyGemTSL It's being caused by line 89 in options.rb

`tasks[:kill_all] = true`

This definition means that docker-compose down and docker-compose rm -vf will be called before the tests start, which will result in all of the associated containers being stopped.

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
retest this please

**Comment by paulrobinson** [comment]
retest this please

**Comment by paulrobinson** [comment]
DCP should have rht_website content now.

**Comment by RubyGemTSL** [comment]
I get no results found now? Fuse, rhel, containers etc return nothing.

**Comment by paulrobinson** [comment]
yeah, the rht_website content still doesn't seem to be available.

**Comment by paulrobinson** [comment]
I have a possible workaround here: https://github.com/redhat-developer/developers.redhat.com/pull/1091

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@wesbos there is still one failure to address on this PR.

Error: expected collection contained: ["Most Recent", "Relevance", "Title"]
actual collection contained: ["Most Recent", "Relevance"]
the missing elements were: ["Title"]

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@wesbos I've raised a PR on your branch with updates to the tests based on comments from Tiffany.

**Comment by unibrew** [comment]
@wesbos or @paulrobinson , Guys do you need those aggregations defined in this search query agg=per_project_counts&agg=tag_cloud&agg=top_contributors&agg=activity_dates_histogram&agg=per_sys_type_counts ? I don't think I see anything on the search page that uses results of those aggregations.
I'm asking because in order to implement some custom boosting of scores as requested in 3 tasks I have I need to reimplement this search as registered query and redoing all aggregations will simply take more time to develop and test.

**Comment by paulrobinson** [comment]
@wesbos, looks like it needs rebasing.

File: _archetypes/errai-archetypes [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 79fc168d19b86fd9bb5e7e7f0b838e14ee04e0c9
+Subproject commit b7669ad1c92554da3b82a8f205dd72f96e145d38
```
Comment (paulrobinson) [inline _archetypes/errai-archetypes]: You need to revert this change. It looks to be breaking the build and it shouldn't be in here.

Comment (wesbos) [inline _archetypes/errai-archetypes]: done

File: _eap-quickstarts [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit d9281fa6c7ca7a498bdf95049c64bbbe41b989cf
+Subproject commit 71a4719e5f52f25857b51b3a4f2e781db5659f45
```
Comment (paulrobinson) [inline _eap-quickstarts]: Please revert this change.

File: javascripts/search.js [hunk 3]
```diff
@@ -1,138 +1,303 @@
...
+    $scope.updateSearch();
+  };
+
+  $scope.updateSearch(); // run on pageload TODO: move to ng-init
+}
+
+
+
+
+// /* global app */
```
Comment (paulrobinson) [inline javascripts/search.js]: Is there a reason why we are keeping this commented out block of code?

Comment (wesbos) [inline javascripts/search.js]: Nope - it's removed in other branch

**Comment by RubyGemTSL** [comment]
The tests are failing due a missing 'end' on line 3 
Please see: https://github.com/wesbos/developers.redhat.com/blob/007ffb74448f56d5d068f2deb4aad2437a397ed1/_cucumber/features/step_definitions/register_steps.rb#L3

**Comment by wesbos** [comment]
Okay fixed it - I tried to merge in master and I must have missed that end block

File: _config/site.yml [hunk 4]
```diff
@@ -141,8 +141,10 @@ profiles:
   # Docker staging configuration (PR and Nightly)
   staging_docker:
     <<: *profile
-    dcp_base_url: http://searchisko:8080/
-    dcp_base_protocol_relative_url: //<%= ENV['ACCESSIBLE_SLAVE_IP'] %>:<%= ENV['SEARCHISKO_HOST_PORT'] %>/
+    dcp_base_url: http://dcp.stage.jboss.org/
+    dcp_base_protocol_relative_url: //dcp.stage.jboss.org/
+#    dcp_base_url: http://searchisko:8080/
+#    dcp_base_protocol_relative_url: //<%= ENV['ACCESSIBLE_SLAVE_IP'] %>:<%= ENV['SEARCHISKO_HOST_PORT'] %>/
```
Comment (paulrobinson) [inline _config/site.yml]: Looks like we are ready to go live with this now. Can you back this out?

Comment (paulrobinson) [inline _config/site.yml]: Actually, hold on please

Comment (paulrobinson) [inline _config/site.yml]: I'll back it out after merging this into master. Otherwise I'll net have a passing build.

File: javascripts/search.js (hunk 3 above)
Comment (paulrobinson) [inline javascripts/search.js]: OK, lets deal with it there then.

**Comment by paulrobinson** [comment]
@wesbos, @RubyGemTSL any reason not to merge this now?

**Comment by wesbos** [comment]
I'm good to merge

**Comment by RubyGemTSL** [comment]
@paulrobinson @wesbos  me too
